packages/shorebird_cli/lib/src/executables/aot_tools.dart

```diff
@@ -98,21 +109,68 @@ class AotTools {
     // distributed as an executable.
     final extension = p.extension(artifactPath);
     if (extension != '.dill' && extension != '.dart') {
-      return process.run(
+      return runFn(
         artifactPath,
         command,
         workingDirectory: workingDirectory,
       );
     }
 
     // local engine versions use .dart and we distribute aot-tools as a .dill
-    return process.run(
+    return runFn(
       shorebirdEnv.dartBinaryFile.path,
       ['run', artifactPath, ...command],
       workingDirectory: workingDirectory,
     );
   }
 
+  /// Similar to [_exec], but logs the sub process stdout and stderr
+  /// as they are emmited.
+  Future<ShorebirdProcessResult> _execWithLiveLogs(
+    List<String> command, {
+    String? workingDirectory,
+  }) {
+    return _exec(
```

Comment:
Not as part of this PR, but I sort of suspect that this will move down a layer to `ShorebirdProcess` as other utilities want the ability to tail stdout/stderr

Reply:
Yes! I hope of doing so when we expand this approach to more sub process. For now I wanted to keep it in here in order to make less changes